src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -68,7 +68,7 @@ impl<'a, 'tcx> Visitor<'tcx> for BorrowckCtxt<'a, 'tcx> {
     }
 
     fn visit_fn(&mut self, fk: FnKind<'tcx>, fd: &'tcx hir::FnDecl,
-                b: hir::ExprId, s: Span, id: ast::NodeId) {
+                b: hir::BodyId, s: Span, id: ast::NodeId) {
```

Comment:
so I can't find a way to make an error out of this (yet), but it seems like we are failing to borrowck embedded bodies -- e.g., in an expression like `[22; X]`, we don't borrowck the expression X, right? The main reason I can't make an error here is that the set of constant expressions we accept in that context is pretty limited, but it seems like it wouldn't necessarily stay that way...

I did run across this, which seems like a bug, but a pre-existing one (this code [compiles on nightly today](https://is.gd/5SDSWz)...but it shouldn't...):

```rust
#![feature(const_fn)]

struct Foo(usize);

const fn get(x: Foo) -> usize {
    x.0
}

const X: Foo = Foo(22);
static Y: usize = get(*&X);

fn main() {
}
```